Sources/AblyChat/DefaultMessages.swift

```diff
@@ -0,0 +1,265 @@
+import Ably
+
+
+public struct MessageEventPayloadWrapper: Hashable, Sendable {
+    let id: String
+    let listener: @Sendable (MessageEventPayload) -> Void
+
+    // Hashable conformance via the id
+    public func hash(into hasher: inout Hasher) {
+        hasher.combine(id)
+    }
+
+    public static func == (lhs: MessageEventPayloadWrapper, rhs: MessageEventPayloadWrapper) -> Bool {
+        return lhs.id == rhs.id
+    }
+    
+    public init(id: String, listener: @Sendable @escaping (MessageEventPayload) -> Void) {
+        self.id = id
+        self.listener = listener
+    }
+}
+
+typealias fromSerial = String
+typealias MessageListeners = [MessageEventPayloadWrapper: fromSerial]
+
+
+public struct MessageEventPayload: Sendable, Hashable {
+    
+    /**
+     * The type of the message event.
+     */
+    let type: MessageEvents
+
+    /**
+     * The message that was received.
+     */
+    public let message: Message
+}
+
+
+actor DefaultMessages: Messages, HandlesDiscontinuity {
+    private let roomID: String
+    internal var channel: (any RealtimeChannelProtocol)?
+    private let chatAPI: ChatAPI
+    private let clientID: String
+    private var listenerSubscriptionPoints: MessageListeners = [:]
+
+    private let lock = NSLock()
+    private let realtime: RealtimeClient
+    
+
+    init(chatAPI: ChatAPI, realtime: RealtimeClient, roomID: String, clientID: String) {
+        self.chatAPI = chatAPI
+        self.roomID = roomID
+        self.clientID = clientID
+        self.realtime = realtime
+        
+        Task {
```

Comment:
Does this work really need to be kicked off by `init` or can it be in response to some user-initiated event? It would be good to avoid creating unstructured tasks like this if possible.

Reply:
I've added a comment, wdyt?

Reply:
Will take another look in the new code and see if I have questions